plugins/linkattacher/src/linkattacherserverimpl.cpp

```diff
@@ -0,0 +1,192 @@
+    #else
+    gazebo::physics::ModelPtr object_model = _world->GetModel(model_name);
+    #endif
+
+    if(!object_model)
+    {
+        yError() << LogPrefix << "Attach: " << model_name << " does not exist in gazebo";
+        return false;
+    }
+    else yInfo() << LogPrefix << "Attach: " << object_model->GetName() << " found";
+
+    gazebo::physics::LinkPtr object_link = object_model->GetLink(model_link_name);
+    if(!object_link)
+    {
+        yError() << LogPrefix << "Attach: " << model_link_name << " is not found";
+    }
+    else yInfo() << LogPrefix << "Attach: " << object_link->GetName() << " found";
+
+    #if GAZEBO_MAJOR_VERSION >= 8
+    gazebo::physics::ModelPtr robot_model = _world->ModelByName(robot_name);
+    #else
+    gazebo::physics::ModelPtr robot_model = _world->GetModel(robot_name);
+    #endif
+
+    if(!robot_model)
+    {
+        yError() << LogPrefix << "Attach: " << robot_name << " does not exist in gazebo";
+        return false;
+    }
+    else yInfo() << LogPrefix << "Attach: " << robot_model->GetName() << " found";
+
+
+    //Get the exact link with only link name instead of full_scoped_link_name
+    gazebo::physics::Link_V robot_model_links = robot_model->GetLinks();
+    for(int i=0; i < robot_model_links.size(); i++)
+    {
+        //E.g iCub::l_hand::l_hand_base_link
+
+        std::string candidate_robot_link_name = robot_model_links[i]->GetScopedName();
+
+        //Display all the links
+        //yInfo() << "Attach --> Full scoped Candidate robot link name: " << candidate_robot_link_name << " found";
+
+        std::size_t lastcolon = candidate_robot_link_name.rfind(":");
+        std::string unscoped_robot_link_name = candidate_robot_link_name.substr(lastcolon+1,std::string::npos);
+        //yInfo() << "Attach --> Unscoped robot link name " << unscoped_robot_link_name;
+
+        if(unscoped_robot_link_name == robot_link_name)
+        {
+            yInfo() << LogPrefix << "Attach: Full scoped Candidate robot link name: " << candidate_robot_link_name << " found";
+            gazebo::physics::LinkPtr robot_link = robot_model_links[i];
+            if(!robot_link)
+            {
+                yError() << LogPrefix << "Attach: " << robot_link_name << " is not found";
+                return false;
+            }
+            else yInfo() << LogPrefix << "Attach: " << robot_link->GetName() << " found";
+
+            //This is joint creation
+            gazebo::physics::JointPtr joint;
+
+            #if GAZEBO_MAJOR_VERSION >= 8
+            joint = _world->Physics()->CreateJoint("fixed",object_model);
+            #else
+            joint = _world->GetPhysicsEngine()->CreateJoint("fixed",object_model);
+            #endif
+
+            if(!joint)
+            {
+                yError() << LogPrefix << "Attach: Unable to create joint";
+                return false;
+            }
+
+            std::string joint_name = model_link_name + "_magnet_joint";
```

Comment:
Please add details about this naming convention `_magnet_joint` used, in the documentation.

Reply:
Nothing specific in choosing the name. It is more distinguishing that using *_fixed_joint*

Reply:
Just so, this information would be useful if user wants to access the properties or this joint.

Reply:
@prashanthr05 updated the documentation with it